llvm/test/Transforms/CorrelatedValuePropagation/ctpop-range.ll

```diff
@@ -0,0 +1,102 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -passes=correlated-propagation %s | FileCheck %s
+
+declare void @use(i1)
+
+define void @ctpop(i8 %v) {
```

Comment:
Can you please split this into four tests, rather than having all of them in one function?